### ISSUE TITLE
Exception: java.lang.NullPointerException: Attempt to read from field 'org.json.JSONObject com.onesignal.UserState.dependValues' on a null object reference

### ISSUE THREAD

**Issue by DimaKoz** [OneSignal/OneSignal-Android-SDK#503]
I got a [crashlog](https://github.com/OneSignal/OneSignal-Android-SDK/files/1914339/log.txt) in OneSignal-Android-SDK 3.8.3
I can't reproduce it.
About devices:
<img width="1089" alt="screen shot 2018-04-16 at 11 07 06" src="https://user-images.githubusercontent.com/3092276/38797306-b8f727aa-4166-11e8-939e-b15a2c35c462.png">

upd:
My initialization of  OneSignal:
```
            if (BuildConfig.DEBUG) {
                OneSignal.setLogLevel(OneSignal.LOG_LEVEL.DEBUG, OneSignal.LOG_LEVEL.WARN);
            } else {
                OneSignal.setLogLevel(OneSignal.LOG_LEVEL.NONE, OneSignal.LOG_LEVEL.NONE);
            }
            OneSignal.startInit(context)
                    .setNotificationOpenedHandler(new NotificationHandler())
                    .autoPromptLocation(false)
                    .inFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification)
                    .unsubscribeWhenNotificationsAreDisabled(true)
                    .disableGmsMissingPrompt(true)
                    .init();

```

**Comment by jkasten2** [comment]
Thanks for the log, contains of it below for reference.

```java
Fatal Exception: java.lang.NullPointerException: Attempt to read from field 'org.json.JSONObject com.onesignal.UserState.dependValues' on a null object reference
       at com.onesignal.UserStateSynchronizer.syncEmailLogout(UserStateSynchronizer.java:152)
       at com.onesignal.UserStateSynchronizer.internalSyncUserState(UserStateSynchronizer.java:164)
       at com.onesignal.UserStateSynchronizer.syncUserState(UserStateSynchronizer.java:157)
       at com.onesignal.OneSignalStateSynchronizer.syncUserState(OneSignalStateSynchronizer.java:70)
       at com.onesignal.OneSignalSyncServiceUtils$SyncRunnable$1.complete(OneSignalSyncServiceUtils.java:220)
       at com.onesignal.LocationGMS.getLocation(LocationGMS.java:137)
       at com.onesignal.OneSignalSyncServiceUtils$SyncRunnable.run(OneSignalSyncServiceUtils.java:225)
       at java.lang.Thread.run(Thread.java:818)
```

We will look into the stack and follow up.

**Comment by rameshvoltella** [comment]
i am also getting new bug after updating to latest version **onesignal 3.8.4**

Affected device :Motorola Moto E (2nd Gen) (otus_ds)

here is the following log 

```
java.lang.NullPointerException: 
 
  at com.onesignal.UserStateSynchronizer.syncEmailLogout (UserStateSynchronizer.java)
 
  at com.onesignal.UserStateSynchronizer.access$200 (UserStateSynchronizer.java)
  or                     .addOnSessionOrCreateExtras (UserStateSynchronizer.java)
  or                     .doCreateOrNewSession (UserStateSynchronizer.java)
  or                     .doEmailLogout (UserStateSynchronizer.java)
  or                     .internalSyncUserState (UserStateSynchronizer.java)
  or                     .isSessionCall (UserStateSynchronizer.java)
 
  at com.onesignal.UserStateSynchronizer.access$400 (UserStateSynchronizer.java)
  or                     .getId (UserStateSynchronizer.java)
  or                     .onSuccessfulSync (UserStateSynchronizer.java)
  or                     .syncUserState (UserStateSynchronizer.java)
 
  at com.onesignal.UserStateSynchronizer$NetworkHandlerThread$1.run (UserStateSynchronizer.java)
 
  at android.os.Handler.handleCallback (Handler.java:751)
 
  at android.os.Handler.dispatchMessage (Handler.java:95)
 
  at android.os.Looper.loop (Looper.java:154)
 
  at android.os.HandlerThread.run (HandlerThread.java:61)
```

**Comment by Nightsd01** [comment]
Hi, thanks for reporting this issue. It should now be fixed (#517).

**Comment by satish123** [comment]
Hi, 

I am also getting the same issue after updating the version 3.10.1

**Find the log**
Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.onesignal.OneSignal$ChangeTagsUpdateHandler.onSuccess(org.json.JSONObject)' on a null object reference
       at com.onesignal.OneSignal$11.run(OneSignal.java:1508)
       at com.onesignal.OneSignal.sendTags(OneSignal.java:1524)
       at com.onesignal.OneSignal.sendTags(OneSignal.java:1450)
       at com.onesignal.OneSignal.sendTag(OneSignal.java:1426)
       at com.app.rehlat.home.ui.FlightSearchActivity.initialize(FlightSearchActivity.java:513)
       at com.app.rehlat.home.ui.FlightSearchActivity.onCreate(FlightSearchActivity.java:341)
       at android.app.Activity.performCreate(Activity.java:7372)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1218)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3147)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3302)
       at android.app.ActivityThread.-wrap12(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1891)
       at android.os.Handler.dispatchMessage(Handler.java:108)
       at android.os.Looper.loop(Looper.java:166)
       at android.app.ActivityThread.main(ActivityThread.java:7425)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)

**Comment by Talkless** [comment]
I have similar issue with `deleteTags()` using `3.10.1`:
```
E/OneSignal(16839): Failed to generate JSON for deleteTags.
E/OneSignal(16839): java.lang.NullPointerException: Attempt to invoke interface method 'void com.onesignal.OneSignal$ChangeTagsUpdateHandler.onSuccess(org.json.JSONObject)' on a null object reference
E/OneSignal(16839): 	at com.onesignal.OneSignal$11.run(OneSignal.java:1508)
E/OneSignal(16839): 	at com.onesignal.OneSignal.sendTags(OneSignal.java:1524)
E/OneSignal(16839): 	at com.onesignal.OneSignal.deleteTags(OneSignal.java:1699)
E/OneSignal(16839): 	at com.onesignal.OneSignal.deleteTags(OneSignal.java:1686)
E/OneSignal(16839): 	at com.mycompany.app.OneSignal.PushNotificationManager.deleteTags(PushNotificationManager.java:79)
E/OneSignal(16839): 	at org.qtproject.qt5.android.QtNative.runPendingCppRunnables(Native Method)
E/OneSignal(16839): 	at org.qtproject.qt5.android.QtNative$1.run(QtNative.java:102)
E/OneSignal(16839): 	at android.os.Handler.handleCallback(Handler.java:739)
E/OneSignal(16839): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/OneSignal(16839): 	at android.os.Looper.loop(Looper.java:211)
E/OneSignal(16839): 	at android.app.ActivityThread.main(ActivityThread.java:5389)
E/OneSignal(16839): 	at java.lang.reflect.Method.invoke(Native Method)
E/OneSignal(16839): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/OneSignal(16839): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
E/OneSignal(16839): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)
```

**Comment by Talkless** [comment]
@satish123 I believe we should create new bug report.

**Comment by jkasten2** [comment]
@satish123 @Talkless Thanks for reporting, this was fixed in PR #611 but hasn't been included in a release yet. It will be rolled into the next one.

In the mean time you can add an empty update handler to avoid the error;
```java
OneSignal.deleteTag("key", new ChangeTagsUpdateHandler() {
   @Override
   public void onSuccess(JSONObject tags) { }
   @Override
   public void onFailure(OneSignal.SendTagsError error) { }
});
```

**Comment by Talkless** [comment]
> It will be rolled into the next one.

Good to know, thanks!
